protocol/kafkaapi/headers/request_header.go

```diff
@@ -0,0 +1,34 @@
+package headers
+
+import (
+	"github.com/codecrafters-io/kafka-tester/protocol/encoder"
+)
+
+// RequestHeader defines the header for a Kafka request
+type RequestHeader struct {
+	// ApiKey defines the API key for the request
+	ApiKey int16
+	// ApiVersion defines the API version for the request
+	ApiVersion int16
+	// CorrelationId defines the correlation ID for the request
+	CorrelationId int32
+	// ClientId defines the client ID for the request
+	ClientId string
+}
+
+// encode v2
+func (h RequestHeader) encode(enc *encoder.Encoder) {
```

Comment:
Let's merge `encode` and `Encode`, the split doesn't really help with readability - also the "encode v2" comment here doesn't make a lot of sense